ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/HealthCheckResource.java

```diff
@@ -70,7 +72,8 @@ public static HealthCheckResource create(
   ) {
     return new HealthCheckResource(
         new HealthCheckAgent(
-            new ServerInternalKsqlClient(ksqlResource, serviceContext),
+            new ServerInternalKsqlClient(ksqlResource,
+                new KsqlSecurityContext(Optional.empty(), serviceContext)),
```

Comment:
The health check does not require any authenticated user to call the KSQL server. An `Optional.empty()` is user is used.